components/ui-api-layer/docs/proposals/exceed-resource-quota.md

```diff
@@ -0,0 +1,189 @@
+
+[ResourceQuota](https://kubernetes.io/docs/concepts/policy/resource-quotas/) provides constraints that limit aggregate resource consumption per Namespace.
+
+The ResourceQuota values can be as follows:
+
+```bash
+Name:            kyma-default
+Resource         Used       Hard
+--------         ---        ---
+limits.memory    7348440Ki  10Gi
+requests.memory  3727576Ki  7Gi
+pods             6          10
+resourcequotas   1          1
+```
+
+The Console calls for the ResourceQuota status. The calls are triggered after: 
+- switching the environment
+- uploading a resource
+- creating a lambda
+- open environment details
+
+There are a few ways to check if the ResourceQuota exceeds its limit.
+
+## Expose the ResourceQuotaStatus
+
+The idea is to create an endpoint which returns the ResourceQuotaStatus.
+ResourceQuotaStatus contains the flag. The value of this flag informs you if the ResourceQuota is exceeded. 
+
+In case when the `used` value equals the `hard` value, the flag is set to true so the ResourceQuota is exceeded.
+
+[Here](https://github.com/dtaylor113/origin-web-console/blob/master/app/scripts/services/quota.js) you can find the example implementation of the checking logic in UI.
+[Here](https://github.com/openshift/origin/blob/master/pkg/quota/apis/quota/types.go) you can find the example of the ResourceQuotaStatus implementation.
+
+### Corner case
+1. The environment has the maximum memory usage set to 1GiB. 
+2. The environment is using 800MiB memory. 
+3. You create a new ReplicaSet/StatefulSet which wants to create 3 new Pods. Single Pod is taking 100MiB of memory. 
+4. You did not exceeded the ResourceQuota memory limit, but the Pod could not create because of lack of memory.
+
+In that situation logic which is comparing `used` and `hard` will not work.
+To secure your logic for that case you need to implement ReplicaSets and StatefulSets checking logic.
+
+### Checking the ReplicaSets:
+
+In this case, the ReplicaSet contains a `failed` condition, which looks as follows:
+
+```yaml
+{
+     "lastTransitionTime": "2018-08-16T09:41:52Z",
+     "message": "pods \"redis-client-6d75bbc666-lrvg6\" is forbidden: exceeded quota: kyma-default, requested: limits.memory=512Mi, used: limits.memory=3Gi, limited: limits.memory=3Gi",
+     "reason": "FailedCreate",
+     "status": "True",
+     "type": "ReplicaFailure"
+}
+```
+You need to check for the `failed` conditions in all ReplicaSets in the given environment to check if the ResourceQuota is exceeded.
+
+### Checking the StatefulSets
+
+StatefulSets do not provide condition like the ReplicaSet.
+In this case you need to check the StatefulSet's Pods statuses.
+
+If at least one of the StatefulSet's Pods have the `ContainerCreating` status, it means that the StatefulSet is still scaling and did not exceed the ResourceQuota.
+You need also check if Pod is in the `Pending` state, it can means that Pod is still creating and it will change its state to `ContainerCreating`.
```

Comment:
From this description I don't know how to write it as a code. Can you be more specific and strict?

Reply:
In my opinion we should now address only problems with Deployment and ReplicaSets. 
Algorithm responsible for checking SS seems to be complicated and requires a lot calls to API Server.